nautobot/core/models/utils.py

```diff
@@ -265,6 +265,13 @@ def construct_natural_slug(values, pk=None):
 def compress_range(iterable):
     """
     Generates compressed range from an expanded range.
+    For example:
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 100, 101, 102, 103, 104, 105, 1000, 1100, 1101, 1102, 1103, 1104, 1105, 1106]
+        =>
+        iter1: (1, 10)
+        iter2: (1000, 1000)
+        iter3: (100, 105)
+        iter4: (1100, 1106)
```

Comment:
Why this ordering?

Reply:
It yields these tuples: 1-10, 1000, 100-105, 1100-1106 ?

Reply:
Do You mean why input iterable is sorted ?

Reply:
Why is (1000, 1000) before (100, 105)?

Reply:
Must be a typo. Sorry I didn't see this one.

Reply:
Fixed